src/ViewRenderer/ApplicationViewInjection.php

```diff
@@ -0,0 +1,57 @@
+<?php
+
+declare(strict_types=1);
+
+namespace App\ViewRenderer;
+
+use Yiisoft\Router\UrlMatcherInterface;
+use Yiisoft\Yii\Web\User\User;
+
+class ApplicationViewInjection implements
+    LayoutParamsInjectionInterface,
+    MetaTagsInjectionInterface,
+    LinkTagsInjectionInterface
+{
+
+    private User $user;
+    private UrlMatcherInterface $urlMatcher;
+
+    public function __construct(
+        User $user,
+        UrlMatcherInterface $urlMatcher
+    ) {
+        $this->user = $user;
+        $this->urlMatcher = $urlMatcher;
+    }
+
+    public function getLayoutParams(): array
+    {
+        return [
+            'user' => $this->user->getIdentity(),
+            'currentUrl' => (string)$this->urlMatcher->getLastMatchedRequest()->getUri(),
+            'brandLabel' => 'Yii Demo',
+        ];
+    }
+
+    public function getMetaTags(): array
+    {
+        return [
+            [
+                '__key' => 'generator',
+                'name' => 'generator',
+                'value' => 'Yii',
+            ],
+        ];
+    }
+
+    public function getLinkTags(): array
+    {
+        return [
+            [
+                '__key' => 'favicon',
```

Comment:
That's static. It's better to have it in HTML directly.

Reply:
I didn't understand... Return HTML, for example '<link rel="icon" type="image/png" href="/myicon.png">' ?

Reply:
I mean there's no reason to inject it. It could be just a line of HTML in layout like https://github.com/yiisoft/yii-demo/blob/master/views/layout/main.php#L32